src/RMGConvertLH5.cc

```diff
@@ -292,19 +292,55 @@ bool RMGConvertLH5::ConvertToLH5Internal() {
     LH5Log(RMGLog::error, "not a remage HDF5 output file (invalid header)?");
     return false;
   }
-  LH5Log(RMGLog::detail, "Opened Geant4 HDF5 file ", fHdf5FileName);
+  LH5Log(RMGLog::detail, "opened Geant4 HDF5 file ", fHdf5FileName);
 
   // rework the ntuples to LGDO tables.
   auto ntuples_group = hfile.openGroup(ntuple_group_name);
   auto ntuples = GetChildren(ntuples_group);
   bool ntuple_success = true;
+  std::string links_group_name = "__links__";
+  std::set<std::string> links;
   for (auto& ntuple : ntuples) {
     if (ntuple.empty()) LH5Log(RMGLog::fatal, "empty ntuple name, how is this possible?");
 
     auto det_group = ntuples_group.openGroup(ntuple);
     ntuple_success &= ConvertNTupleToTable(det_group);
     det_group.close();
 
+    // Check fNtupleMeta object for an entry whose second field matches the current ntuple name.
+    for (const auto& item : fNtupleMeta) {
+      // item is a pair: item.first is an int, item.second is a std::pair<int, std::string>
+      if (item.second.second == ntuple) {
+        // create group named "links"
+        if (!ExistsByType(ntuples_group, links_group_name, H5O_TYPE_GROUP)) {
+          auto links_group = ntuples_group.createGroup(links_group_name);
+          links_group.close();
+        }
+
+        // form soft link name "detUID" where UID is item.second.first.
+        auto soft_link_name = "uid" + std::to_string(item.second.first);
+        auto soft_link_name_full = ntuple_group_name + "/" + links_group_name + "/" + soft_link_name;
+        // do not create if the soft link already exists.
+        if (!ntuples_group.nameExists(soft_link_name)) {
+          // create a soft link to the current group itself.
+          herr_t err = H5Lcreate_soft(
+              ("/" + ntuple_group_name + "/" + ntuple).c_str(),
```

Comment:
there should be a c++ api for this, I guess? Such as `H5Location::link(...)`:

https://github.com/HDFGroup/hdf5/blob/2741b4079d07ec7016c6855a7e70a5e6af0099f4/c%2B%2B/src/H5Location.cpp#L1001-L1042

important note: you need to use the variant with two string arguments, other overloads create hard links...

wit this we can avoid the `c_str()`s and the explicit error handling

Reply:
I guess this is the best:

https://github.com/HDFGroup/hdf5/blob/2741b4079d07ec7016c6855a7e70a5e6af0099f4/c%2B%2B/src/H5Location.cpp#L1191

Reply:
certainly yes